### PR TITLE
Pages: Fix blog posts indicator when setting posts page to homepage

### PR DESCRIPTION
**Pull request by jonathansadowski**
This PR fixes a bug where the blog posts indicator was incorrect when setting a static posts page to the homepage.

To reproduce the bug:

* Set both a static front-page and a static posts page from wp-admin: `/wp-admin/options-reading.php`
image

* Reload the pages page for that site in Calypso. It should look like this:
image

* Set the page being subsumed by the "Blog Posts" to the homepage:
image

* Observe that the state is inconsistent: the page is listed as both the static home page and "Your latest posts"
image

* Reload the page and observe that it reconciles:
image

Checkout this branch and verify that the bug is fixed.

### CODE REVIEW

File: client/state/sites/actions.js [hunk 1]
```diff
@@ -115,13 +115,22 @@ export function setFrontPage( siteId, pageId, successCallback ) {
...
 		};
 
-		return wpcom.undocumented().setSiteHomepageSettings( siteId, requestData ).then( () => {
+		return wpcom.undocumented().setSiteHomepageSettings( siteId, requestData ).then( ( response ) => {
+			const updatedOptions = {
+				page_on_front: parseInt( response.page_on_front_id, 10 ),
+				show_on_front: response.is_page_on_front ? 'page' : 'posts',
+			};
+
+			if ( response.page_for_posts_id ) {
```
Comment (mattsherman) [inline client/state/sites/actions.js]: Even if `page_for_posts_id` is zero it should be set. In fact, this is what happens when I try and verify the fix... setting the posts page to the homepage results in this being zero.

**Comment by jonathansadowski** [comment]
@mattsherman I've updated to account for that case.

**Comment by mattsherman** [comment]
Re-tested and works as expected. 👍

**Comment by jonathansadowski** [comment]
Thanks for the review!

**Pull request merged**